core/server/worker/src/test/java/alluxio/worker/block/BlockMasterClientTest.java

```diff
@@ -80,23 +79,26 @@
 
 public class BlockMasterClientTest {
 
+  // alert: you CANNOT just set TEST_SOCKET_ADDRESS to
+  // configuration, because it will get turned to string "localhost/127.0.0.1:9999"
+  // This string cannot be parsed accurately by the current parsing algorithm
   private static final InetSocketAddress TEST_SOCKET_ADDRESS =
-      new InetSocketAddress("0.0.0.0", 9999);
+      new InetSocketAddress("localhost", 9999);
+  private static final String TEST_SOCKET_ADDRESS_STRING = "localhost:9999";
```

Comment:
nit: does `TEST_SOCKET_ADDRESS.toString()` work here? If so, we can get rid of this additional constant.

Reply:
It doesn't work here for the reason stated in the comment🤦‍♂️. The `InetSocketAddress` will be turned into "localhost/127.0.0.1:9999". This string cannot be parsed correctly by the current algorithm: [NetworkAddressUtils#parseInetSocketAddress](https://github.com/Alluxio/alluxio/blob/master/core/common/src/main/java/alluxio/util/network/NetworkAddressUtils.java).

Reply:
I tried some alternatives like `new InetSocketAddress("127.0.0.1", 9999)` and `new InetSocketAddress("0.0.0.0", 9999)`. They don't seem to produce connectable addresses because the constructor expects a host name in that argument.

Reply:
wow interesting, do you think [NetworkAddressUtils#parseInetSocketAddress](https://github.com/Alluxio/alluxio/blob/master/core/common/src/main/java/alluxio/util/network/NetworkAddressUtils.java) can be fixed easily? If not, we can leave it be.

I'm fine with keeping all necessary variables here. It's just the comment seems a bit verbose and distractive.

Reply:
> wow interesting, do you think [NetworkAddressUtils#parseInetSocketAddress](https://github.com/Alluxio/alluxio/blob/master/core/common/src/main/java/alluxio/util/network/NetworkAddressUtils.java) can be fixed easily? If not, we can leave it be.
> 
> I'm fine with keeping all necessary variables here. It's just the comment seems a bit verbose and distractive.

I would say yes. We could fix it by splitting the hostname by "/" and take the first part.

Reply:
Since "/" is not a valid character in a hostname, this is not likely to break present working hostnames.